### PR TITLE
Add option for publishing to MQTT broker

### PR DESCRIPTION
**Pull request by tsundvoll**

### CODE REVIEW

File: src/isar/services/service_connections/mqtt/mqtt_client.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        self.client.on_disconnect = self.on_disconnect
+
+        self.client.username_pw_set(username=username, password=password)
+        self.client.connect(host=host, port=port, keepalive=60)
+
+    def on_connect(self, client, userdata, flags, rc):
+        self.logger.info("Connection returned result: " + mqtt.connack_string(rc))
+
+    def on_disconnect(self, client, userdata, rc):
+        # Add backoff strategy, retry after 5, 10, 30, 60 sec for example
```
Comment (tsundvoll) [inline src/isar/services/service_connections/mqtt/mqtt_client.py:64]: Should we remove this comment and add it as an issue?

**Review by tsundvoll — COMMENTED** [review]
Only one comment. Apart from that it looks good to me. Can't approve, since it is originally my PR :sweat_smile:

File: src/isar/services/service_connections/mqtt/mqtt_client.py (hunk 1 above)
Comment (aestene) [reply src/isar/services/service_connections/mqtt/mqtt_client.py:64]: Yes, that's a good idea :)

File: src/isar/state_machine/states/send.py [hunk 2]
```diff
@@ -85,6 +94,10 @@ def _run(self):
 
             if send_success:
                 self.state_machine.current_task.status = TaskStatus.Scheduled
+
+                if self.state_machine.mqtt_client:
+                    self.state_machine.publish_task_status()
+
```
Comment (aestene) [inline src/isar/state_machine/states/send.py:100]: Remove this

File: src/isar/services/service_connections/mqtt/mqtt_client.py (hunk 1 above)
Comment (aestene) [reply src/isar/services/service_connections/mqtt/mqtt_client.py:64]: #198

File: src/isar/state_machine/states/send.py [hunk 3]
```diff
@@ -34,9 +34,17 @@ def start(self):
         self.state_machine.update_current_task()
         self.logger.info(f"State: {self.state_machine.current_state}")
 
+        if self.state_machine.mqtt_client:
+            self.state_machine.publish_task_status()
+            self.state_machine.publish_mission()
```
[the remark this answers is not in the document]
Comment (aestene) [reply src/isar/state_machine/states/send.py:39]: Since the mock makes the `mqtt_client` variable not `None` this actually works out well already.

**Pull request merged**